features/XouBaseApiLoginAndRegister/Repo/UserModel.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  UserModel.swift
+//  XouDevSpec
+//
+//  Created by Xander Schoeman on 2020/04/19.
+//
+
+import Foundation
+
+public struct User: Decodable {
+    public var id: Int?
+    public  var username: String?
```

Comment:
Do all of these really need to be optional, may result in more handling that you need. For example, id, username, email, etc may be fields that will never be null and are necessary for your code.
If you allow them to be null, it will never fail decoding even if the values aren't there and so you have to handle checking it for null elsewhere. 

It makes more sense to instead make what can be non-nullable as such  so you can handle missing fields immediately when you get them. Just my opinion, you may be handling it in some other way already

Reply:
That does make more sense, I will change them now.